src/uu/du/src/du.rs

```diff
@@ -278,8 +614,46 @@ fn du(
         'file_loop: for f in read {
             match f {
                 Ok(entry) => {
-                    match Stat::new(&entry.path(), Some(&entry), options) {
+                    let entry_path = entry.path();
+
+                    // Check if this is a symlink when using -L
+                    let mut current_symlink_depth = symlink_depth;
+                    let is_symlink = match entry.file_type() {
+                        Ok(ft) => ft.is_symlink(),
+                        Err(_) => false,
+                    };
```

Comment:
I think you can simplify it with something like:
```suggestion
                    let is_symlink = entry.file_type().map_or(false, |ft| ft.is_symlink());
```